File: apps/java-sdk/src/main/java/com/firecrawl/models/AgentSnapshotResponse.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Response from the agent snapshot endpoint.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class AgentSnapshotResponse {

    private boolean success;
    private String id;
    private String snapshotId;
    private String snapshot;
    private String error;

    public boolean isSuccess() { return success; }
    public String getId() { return id; }
    public String getSnapshotId() { return snapshotId; }
    public String getSnapshot() { return snapshot; }
    public String getError() { return error; }

    @Override
    public String toString() {
        return "AgentSnapshotResponse{success=" + success + ", id=" + id + ", snapshotId=" + snapshotId + "}";
    }
}

File: apps/ruby-sdk/lib/firecrawl/models/agent_snapshot_response.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # Snapshot response for an agent task.
    class AgentSnapshotResponse
      attr_reader :success, :id, :snapshot_id, :snapshot, :error

      def initialize(raw)
        @success = raw["success"]
        @id = raw["id"]
        @snapshot_id = raw["snapshotId"]
        @snapshot = raw["snapshot"]
        @error = raw["error"]
      end

      def to_s
        "AgentSnapshotResponse{id=#{id}, snapshot_id=#{snapshot_id}, success=#{success}}"
      end
    end
  end
end
